### PR TITLE
Mutex around parse

### PR DESCRIPTION
**Pull request by yarikoptic**
Portion from #194 which is trivial and independent from that one which might take longer to fixup properly.

### CODE REVIEW

File: linkcheck/parser/__init__.py [hunk 1]
```diff
@@ -125,7 +130,9 @@ def find_links (url_data, callback, tags):
     handler.parser = parser
     # parse
     try:
-        parser.feed(url_data.get_content())
+        content = url_data.get_content()
+        with parse_mutex:
+            parser.feed(content)
         parser.flush()
```
Comment (mgedmin) [inline linkcheck/parser/__init__.py]: I may be missing previous context, but it's unclear to me why `parser.feed()` is not thread-safe, but `parser.flush()` is fine?

Are we using the same parser instance in multiple threads?  (It doesn't look like we do.)

Is the parser using global state shared between multiple parser instances?

Is it perhaps GNU bison that's giving us a thread-unsafe parser?

Perhaps we should prioritize #119 that replaces our homegrown C parser with BeatifulSoup?

**Comment by yarikoptic** [comment]
pushed that little patch offsetting flush to be under mutex as well.

Would be nice if there was a test demonstrating its need... unfortunately the one targeting anchors being added in #194 is not triggering it AFAIK, and originally I detected it while running on a ad-hoc test case I simulated in local collection of .htmls I generated by building https://github.com/bids-standard/bids-specification .

**Comment by yarikoptic** [comment]
> I have no time/inclination to study the C sources generated by bison

who would? ;-)
Unfortunately I've not kept an original ad-hoc modified sample to test anchors situation and ATM without it cannot reproduce fluctuating reported links count to provide you undeniable evidence of positive effects.

**Comment by yarikoptic** [comment]
what should be done to get at least this one merged? ;-)

**Comment by mgedmin** [comment]
I would like to hear @anarcat's opinion.

**Comment by anarcat** [comment]
@mgedmin i'll let you do the honors considering you actually herded this considerably more than i did. :)

**Comment by mgedmin** [comment]
Ha!  You just want me to feel responsible if this ever backfires ;)

**Comment by anarcat** [comment]
yes, of course, that too. :p

**Comment by yarikoptic** [comment]
Thank you guys!  I still hope some time to get back to #194 -- feel welcome to beat me to it ;-)

**Pull request merged**